### DISCUSSION TITLE
Issue finding keys for Kindle for Android books

### DISCUSSION THREAD

**Discussion by abbotm** [noDRM/DeDRM_tools#316] [General]
The DeDRM 7.2.1plugin isn't recognizing the BACKUP.AB file created using ADB.EXE (Version 1.0.41 Version 34.0.1-9680074) as a valid backup file. Trying to debug this, I see that the BACKUP.AB file is 47 bytes long. That doesn't seem correct.
Are there older/better versions of ADB.EXE Backup that are known to work? or some other utility that can make a backup file that is compatible with DeDRM? Are there command line switches I should be using with the ADB backup command?

Help would be much appreciated. 
Thx

**Comment by ElleKayEm** [comment]
You'll need an older version of Kindle for Android to get a usable version of the backup file.  See this post:  https://www.mobileread.com/forums/showthread.php?t=352278

Also if you are on calibre 6, update to DeDRM 10.0.3:  https://github.com/noDRM/DeDRM_tools/releases/tag/v10.0.3

**Reply by abbotm** [reply]
Took me a couple of tries, but this method worked!!
Follow all the instructions exactly, and use the recommended version of Kindle for Android.

Many thanks to ElleKayEm.

**Reply by ElleKayEm** [reply]
Thanks for reporting back that this method is continuing to work.
